### ISSUE TITLE
config file moving aid

### ISSUE THREAD

**Issue by lnussel** [rpm-software-management/rpm#1296]
There are efforts to get rid of distro default config files in /etc in favor of built in defaults or files in eg /usr/etc. Means rpms no longer install stuff into /etc. The transition there however requires some handling of existing files in /etc. Unmodified ones should get removed while modified ones need to stay in place. Right now on openSUSE side we have an advice in the wiki to add [scriptlets](https://en.opensuse.org/openSUSE:Packaging_UsrEtc#Moving_of_configuration_files) in %pre and %posttrans. Not exactly convenient nor smart.

Also, that model assumes that such packages have no mention of the config files in /etc at all anymore. IMHO it would still be desirable own the ones in /etc though. So if one runs `rpm -qc vim` one would see `/etc/vimrc` even if that file isn't shipped.

So assuming the old package had

    %config(noreplace) /etc/vimrc
The new one should have something like

    %config(missingok) /etc/vimrc
or

    %ghost %config /etc/vimrc

Currently however the first version would lead to actually installing a (potentially zero sized) file in /etc, and the %ghost form would not remove an unmodified config on update.

Any thoughts on this?

**Comment by pmatilai** [comment]
Yep, rpm has no means to track content moving around except through symlinks.
I can see how it would be useful of course, but it'd require somehow recording such transformations in the package, and teaching the rest of rpm to honor those.

Just a wild idea: perhaps file moving could be encoded in the file list as a virtual symlink (call it %alias or something) and have the fingerprinting code follow these. Something in that ballpark *may* be somehow feasible because it'd at least be basing of something rpm already knows.

**Comment by pmatilai** [comment]
@ffesti pointed out that a fitting terminology for this would be file-level obsoletes.
